src/fstab-generator/fstab-generator.c

```diff
@@ -164,6 +164,29 @@ static int write_idle_timeout(FILE *f, const char *where, const char *opts) {
         return 0;
 }
 
+static int write_mount_timeout(FILE *f, const char *where, const char *opts) {
+        _cleanup_free_ char *timeout = NULL;
+        char timespan[FORMAT_TIMESPAN_MAX];
+        usec_t u;
+        int r;
+
+        r = fstab_filter_options(opts, "x-systemd.mount-timeout\0", NULL, &timeout, NULL);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to parse options: %m");
+        if (r == 0)
+                return 0;
+
+        r = parse_sec(timeout, &u);
+        if (r < 0) {
+                log_warning("Failed to parse timeout for %s, ignoring: %s", where, timeout);
+                return 0;
+        }
+
+        fprintf(f, "TimeoutSec=%s\n", format_timespan(timespan, sizeof(timespan), u, 0));
```

Comment:
So we have two almost identical functions, differing only in two strings. You should extract most of the code to a helper function, and have two stub functions which only differ in those two strings, or simply provide those arguments in the call place.